kiosk/sources/extensions/owned_kiosk.move

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Description:
+///
+module kiosk::owned_kiosk {
+    use std::option::{Self, Option};
+    use sui::transfer;
+    use sui::kiosk::{Self, Kiosk, KioskOwnerCap};
+    use sui::object::{Self, ID, UID};
+    use sui::tx_context::{sender, TxContext};
+    use sui::dynamic_field as df;
+
+    const EIncorrectCapObject: u64 = 0;
+    const EIncorrectOwnedObject: u64 = 1;
+
+    /// A key-only wrapper for the KioskOwnerCap. Makes sure that the Kiosk can
+    /// not be traded altogether with its contents.
+    struct OwnedKiosk has key {
+        id: UID,
+        cap: Option<KioskOwnerCap>
+    }
+
+    /// The hot potato making sure the KioskOwnerCap is returned after borrowing.
+    struct Borrow { cap_id: ID, owned_id: ID }
+
+    /// The dynamic field to mark the Kiosk as owned (to allow guaranteed owner
+    /// checks through the Kiosk).
+    struct OwnerMarker has copy, store, drop {}
+
+    /// Wrap the KioskOwnerCap making the Kiosk "owned" and non-transferable.
+    public fun new(kiosk: &mut Kiosk, cap: KioskOwnerCap, ctx: &mut TxContext) {
+        let owner = sender(ctx);
+
+        // set the owner property of the Kiosk
+        kiosk::set_owner(kiosk, &cap, ctx);
+
+        // add the owner marker to the Kiosk; uses `_as_owner` to always pass,
+        // even if Kiosk "allow_extensions" is set to false
+        df::add(
+            kiosk::uid_mut_as_owner(kiosk, &cap),
+            OwnerMarker {},
+            owner
+        );
+
+        // wrap the Cap in the OwnedKiosk
+        transfer::transfer(OwnedKiosk {
```

Comment:
Return rather than transfer for PTB-friendliness?

Reply:
It's a `key` only asset. One way out is to add a custom `transfer_to_sender` function for scenario when an "owned" version is installed silently.